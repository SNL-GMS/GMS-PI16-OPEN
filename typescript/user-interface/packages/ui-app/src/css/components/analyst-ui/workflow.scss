@import '../../gms-colors.scss';
@import '../../gms-fonts.scss';

$BLOCK_HEIGHT_PX: 24px;
$ROW_COMPUTED_HEIGHT_PX: 26px;
$BLOCK_WIDTH_PX: 140px;
$TABLE_MARGIN: 34px;
$DIVIDER_MARGIN: 31px;
//The table margin plus some manual fiddling
$WIDTH_OF_LEFT_CURTAIN: 43px;
$gap-between-rows: 0.75em;

.gms-workflow-time-axis {
  path,
  line {
    fill: none;
    stroke: $gms-prominent;
    shape-rendering: crispEdges;
  }
  text {
    font-family: $gms-sans;
    font-size: 12px;
    fill: $gms-main;
    &.day-label {
      stroke: $workflow-day-divider;
      font-size: 13px;
    }
  }
}

.workflow-display-window {
  height: 100%;
}

.workflow-panel {
  height: 100%;
  width: 100%;
}

.workflow-scroll-wrapper {
  max-height: calc(100% - 55px);
  overflow-y: scroll;
}

.interval-table-wrapper {
  flex: 1 0 auto;
  display: flex;
  margin-left: $TABLE_MARGIN;
  margin-right: $BLOCK_WIDTH_PX;
  flex-direction: column;
  overflow-x: scroll;
  overflow-y: hidden;
  position: static;
  gap: 2px;
}

.interval-table-wrapper-wrapper {
  position: relative;
  overflow-y: hidden;
  overflow-x: hidden;
}

// Left is set to account for the workflow-table-wrapper's left margin, minus the size of the button
.stage-row__expand-button {
  height: $BLOCK_HEIGHT_PX;
  border: 1px solid $gms-recessed;
  width: $BLOCK_HEIGHT_PX;
  transition: margin-bottom 0.15s;
  min-height: $ROW_COMPUTED_HEIGHT_PX;
  min-width: $ROW_COMPUTED_HEIGHT_PX;
  padding: 0;
  position: absolute;
  left: 7px;
  margin-top: -1px;
  z-index: 5;
}

.stage-row {
  display: flex;
  gap: 2px;
}

.stage-column {
  display: flex;
  flex-direction: column;
  gap: 2px;
}

.internal-row {
  display: grid;
  grid-auto-columns: 8px;
  gap: 2px;
}

.workflow-tooltip-container {
  display: grid;
  grid-template-columns: 2fr 3fr;
  margin: 0 auto;
  gap: 0.25em;
  padding-bottom: 0.5em;
  border-bottom: 1px dashed $gms-soft;
}

.workflow-tooltip-container .label-value__label {
  font-weight: bold;
}

.workflow-tooltip-container:last-child {
  border-bottom: none;
  padding-bottom: 0;
}

.workflow-tooltip-container + .workflow-tooltip-container {
  padding-top: 0.5em;
}

// The absolute position lets it float to the left over the workflow display
// The margin right and top are for alignment
.workflow-table-label {
  height: $ROW_COMPUTED_HEIGHT_PX;
  width: $BLOCK_WIDTH_PX;
  border: 1px solid;
  display: flex;
  border-color: $gms-background;
  justify-content: center;
  align-items: center;
  background: $gms-soft;
  flex: 0 0 auto;
  position: absolute;
  right: 0px;
  margin-right: 4px;
  margin-top: -1px;
  z-index: 5;
}

.workflow-table-label--activity {
  filter: $interval-activity-darken-cell-by;
}

.workflow-row {
  position: relative;
  display: flex;
  margin-bottom: 2px;
}

.interval-cell {
  display: flex;
  flex-direction: column;
  height: 100%;
  height: $BLOCK_HEIGHT_PX;
  border: 1px solid $gms-background;
  align-items: center;
  justify-content: center;
  margin: 0;
  gap: 2px;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  .percent-bar {
    height: 100%;
    width: 100%;
  }
}

.interval-cell:hover {
  filter: brightness(1.2);
}

.interval-cell--in-progress {
  background-color: $interval-in-progress-fill;
  border: 1px solid $interval-in-progress-border;
  .percent-bar {
    background-color: scale-color(
      $color: $interval-in-progress-fill,
      $lightness: 25%,
      $saturation: -15%
    );
  }
}

.interval-cell--complete {
  background-color: $interval-complete-fill;
  border: 1px solid $interval-complete-border;
}

.interval-cell--not-complete {
  background-color: $interval-not-complete-fill;
  border: 1px solid $interval-not-complete-border;
}

.interval-cell--not-started {
  background-color: $interval-not-started-fill;
  border: 1px solid $interval-not-started-border;
}

.interval-cell--skipped {
  background-color: $interval-skipped-fill;
  border: 1px solid $interval-skipped-border;
}

.interval-cell--failed {
  background-color: $interval-failed-fill;
  border: 1px solid $interval-failed-border;
}

.interval-cell--selected {
  border: 2px solid $gms-selection;
}

.interval-cell--activity-cell {
  filter: $interval-activity-darken-cell-by;
}

.interval-cell--activity-cell:hover {
  filter: brightness(1);
}

.interval-cell--stale {
  filter: brightness(0.5);
}

.workflow-day-divider {
  position: absolute;
  top: 0px;
  bottom: 0px;
  width: 4px;
  z-index: 1;
  margin-left: $DIVIDER_MARGIN;
}

// To account for the offset of the workflow-table, the time axis needs to be moved to left by 1
.time-axis-wrapper {
  margin-left: -1px;
  width: calc(100%-131px);
  height: 25px;
  flex: 0 0 auto;
  position: static;
  z-index: 1;
}

.time-axis {
  width: 100%;
  height: 25px;
  overflow: hidden;
}

.interval-table__curtain-left {
  width: $WIDTH_OF_LEFT_CURTAIN;
  position: absolute;
  background-color: $gms-background;
  height: 100vh;
  z-index: 2;
  left: -6px;
}

.workflow-ellipsis {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100px;
  text-align: center;
}

.workflow-open-anything-row {
  display: flex;
  flex-flow: wrap;
  box-sizing: border-box;
  align-items: center;
  min-width: 56ch;
  & > * {
    margin-left: 0.5rem;
  }
}

.open-anything-contents .dropdown-container {
  display: flex;
  justify-content: space-between;
}

.workflow-open-anything-row .dropdown-container .span {
  display: flex;
}

.open-anything-contents .date-range-picker {
  width: 100%;
  display: flex;
  margin-bottom: 5px;
}

.open-anything-contents .date-range-picker .date-range-picker__range-input {
  width: 65%;
  margin: 0px;
}

.open-anything-contents
  .date-range-picker
  .date-range-picker__range-input
  .bp3-popover-target
  .bp3-control-group
  * {
  flex-shrink: 1;
}

.open-anything-contents .date-range-picker .date-range-picker__label {
  font-size: 1.125em;
  font-weight: 600;
  margin: 0px;
  width: 35%;
}

.open-anything-contents .dropdown-label {
  font-size: 1.125em;
  font-weight: 600;
  margin: 0px;
  width: 35%;
}

.open-anything-contents .dropdown-selector {
  margin-bottom: 5px;
  width: 65%;
}

.interval-confirmation-contents {
  display: flex;
}

.interval-confirmation-text {
  width: 90%;
}

.interval-confirmation-icon {
  width: 10%;
}

.interval-confirmation-header {
  font-weight: bold;
}

.open-anything-error {
  color: $gms-strong-warning;
  display: flex;
  min-height: 20px;
}
.open-anything-error-text {
  margin-left: 5px;
  width: calc(100%-20px);
}
