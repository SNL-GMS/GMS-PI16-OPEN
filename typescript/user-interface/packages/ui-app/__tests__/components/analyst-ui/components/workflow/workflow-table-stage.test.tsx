/* eslint-disable react/jsx-props-no-spreading */
import { createStore } from '@gms/ui-state';
import * as Enzyme from 'enzyme';
import * as React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';

import { WorkflowTableStageProps } from '../../../../../src/ts/components/analyst-ui/components/workflow/types';
import { WorkflowContext } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-context';
import { WorkflowTableStage } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-table-stage';
import * as WorkFlowDataTypes from './workflow-data-types';

// set up window alert and open so we don't see errors
window.alert = jest.fn();
window.open = jest.fn();

window.ResizeObserver = jest.fn(() => {
  return { observe: jest.fn(), disconnect: jest.fn(), unobserve: jest.fn() };
});
const globalAny: any = global;
globalAny.ResizeObserver = window.ResizeObserver;
globalAny.DOMRect = jest.fn(() => ({}));

const MOCK_TIME = 1606818240000;
global.Date.now = jest.fn(() => MOCK_TIME);

const TIME_TO_WAIT_MS = 200;
const store: any = createStore();

/**
 * Fixes React warning that "An update to Component inside a test was not wrapped in act(...)."
 */
const waitForComponentToPaint = async (wrapper: any): Promise<void> => {
  // eslint-disable-next-line @typescript-eslint/await-thenable
  await act(async () => {
    await new Promise(resolve => setTimeout(resolve, TIME_TO_WAIT_MS));
    wrapper.update();
  });
};

describe('Workflow Table Stage', () => {
  it('is exported', () => {
    expect(WorkflowTableStage).toBeDefined();
  });

  const workflowTableStageProps: WorkflowTableStageProps = {
    stage: WorkFlowDataTypes.interactiveAnalysisStageInterval,
    isExpanded: false,
    subRowNames: []
  };

  const rowNames = ['row1', 'row2'];

  const workflowTableStageProps2: WorkflowTableStageProps = {
    stage: WorkFlowDataTypes.interactiveAnalysisStageInterval,
    isExpanded: false,
    subRowNames: rowNames
  };
  const workflowTableStageProps3: WorkflowTableStageProps = {
    stage: WorkFlowDataTypes.automaticProcessingStageInterval,
    isExpanded: false,
    subRowNames: []
  };

  const workflowTableStageProps4: WorkflowTableStageProps = {
    stage: WorkFlowDataTypes.automaticProcessingStageInterval,
    isExpanded: false,
    subRowNames: rowNames
  };

  it('matches snapshot', () => {
    const component = Enzyme.mount(
      <Provider store={store}>
        <WorkflowContext.Provider
          value={{
            staleStartTime: 1,
            allActivitiesOpenForSelectedInterval: false,
            openConfirmationPrompt: jest.fn(),
            openAnythingConfirmationPrompt: jest.fn(),
            closeConfirmationPrompt: jest.fn()
          }}
        >
          <WorkflowTableStage {...workflowTableStageProps} />
        </WorkflowContext.Provider>
      </Provider>
    );
    expect(component).toMatchSnapshot();
    const workflowTableStage: any = component.find('WorkflowTableStage').instance();
    expect(workflowTableStage).toBeDefined();
  });

  it('shallow mounts', () => {
    const shallow = Enzyme.shallow(
      <Provider store={store}>
        <WorkflowContext.Provider
          value={{
            staleStartTime: 1,
            allActivitiesOpenForSelectedInterval: false,
            openConfirmationPrompt: jest.fn(),
            openAnythingConfirmationPrompt: jest.fn(),
            closeConfirmationPrompt: jest.fn()
          }}
        >
          <WorkflowTableStage {...workflowTableStageProps} />{' '}
        </WorkflowContext.Provider>
      </Provider>
    );
    expect(shallow).toMatchSnapshot();
    const workflowTableStage: any = shallow.find('WorkflowTableStage');
    expect(workflowTableStage).toBeDefined();
  });

  it('Workflow Table Stage functions and clicks', async () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <WorkflowContext.Provider
          value={{
            staleStartTime: 1,
            allActivitiesOpenForSelectedInterval: false,
            openConfirmationPrompt: jest.fn(),
            openAnythingConfirmationPrompt: jest.fn(),
            closeConfirmationPrompt: jest.fn()
          }}
        >
          <WorkflowTableStage {...workflowTableStageProps} />{' '}
        </WorkflowContext.Provider>
      </Provider>
    );

    expect(wrapper.props().isExpanded).toBeFalsy();

    wrapper.setProps({ isExpanded: true });

    await waitForComponentToPaint(wrapper);
    expect(wrapper.props().isExpanded).toBeTruthy();
  });

  it('mounts with automatic processing interval', () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <WorkflowContext.Provider
          value={{
            staleStartTime: 1,
            allActivitiesOpenForSelectedInterval: false,
            openConfirmationPrompt: jest.fn(),
            openAnythingConfirmationPrompt: jest.fn(),
            closeConfirmationPrompt: jest.fn()
          }}
        >
          <WorkflowTableStage {...workflowTableStageProps3} />{' '}
        </WorkflowContext.Provider>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Workflow Table Stage handles subRowNames', async () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <WorkflowContext.Provider
          value={{
            staleStartTime: 1,
            allActivitiesOpenForSelectedInterval: false,
            openConfirmationPrompt: jest.fn(),
            openAnythingConfirmationPrompt: jest.fn(),
            closeConfirmationPrompt: jest.fn()
          }}
        >
          <WorkflowTableStage {...workflowTableStageProps2} />{' '}
        </WorkflowContext.Provider>
      </Provider>
    );

    wrapper.update();
    expect(wrapper.props().isExpanded).toBeFalsy();

    wrapper.setProps({ isExpanded: true });

    await waitForComponentToPaint(wrapper);
    expect(wrapper.props().isExpanded).toBeTruthy();

    wrapper.setProps(workflowTableStageProps3);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.props().isExpanded).toBeFalsy();

    wrapper.setProps(workflowTableStageProps4);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.props().isExpanded).toBeFalsy();
  });
});
