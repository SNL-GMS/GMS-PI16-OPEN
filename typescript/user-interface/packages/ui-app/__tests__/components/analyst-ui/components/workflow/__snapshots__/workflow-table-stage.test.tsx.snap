// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow Table Stage matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <WorkflowTableStage
    isExpanded={false}
    stage={
      Object {
        "activityIntervals": Array [
          Object {
            "activeAnalysts": Array [
              "larry",
              "moe",
              "curly",
            ],
            "comment": "interval example",
            "endTime": 1622057187,
            "modificationTime": 1622057667,
            "name": "Event Review",
            "percentAvailable": 100,
            "processingEndTime": 1622057547,
            "processingStartTime": 1622057487,
            "stageName": "AL1",
            "startTime": 1622053587,
            "status": "IN_PROGRESS",
            "storageTime": 1622057607,
          },
        ],
        "comment": "interval example",
        "endTime": 1622057187,
        "modificationTime": 1622057667,
        "name": "AL1",
        "percentAvailable": 100,
        "processingEndTime": 1622057547,
        "processingStartTime": 1622057487,
        "stageMetrics": Object {
          "associatedSignalDetectionCount": 34,
          "eventCount": 21,
          "maxMagnitude": 8,
          "unassociatedSignalDetectionCount": 55,
        },
        "stageMode": "INTERACTIVE",
        "startTime": 1622053587,
        "status": "IN_PROGRESS",
        "storageTime": 1622057607,
      }
    }
    subRowNames={Array []}
  >
    <div
      className="stage-column"
    >
      <Tooltip
        activeAnalysts={
          Array [
            "larry",
            "moe",
            "curly",
          ]
        }
        interval={
          Object {
            "activityIntervals": Array [
              Object {
                "activeAnalysts": Array [
                  "larry",
                  "moe",
                  "curly",
                ],
                "comment": "interval example",
                "endTime": 1622057187,
                "modificationTime": 1622057667,
                "name": "Event Review",
                "percentAvailable": 100,
                "processingEndTime": 1622057547,
                "processingStartTime": 1622057487,
                "stageName": "AL1",
                "startTime": 1622053587,
                "status": "IN_PROGRESS",
                "storageTime": 1622057607,
              },
            ],
            "comment": "interval example",
            "endTime": 1622057187,
            "modificationTime": 1622057667,
            "name": "AL1",
            "percentAvailable": 100,
            "processingEndTime": 1622057547,
            "processingStartTime": 1622057487,
            "stageMetrics": Object {
              "associatedSignalDetectionCount": 34,
              "eventCount": 21,
              "maxMagnitude": 8,
              "unassociatedSignalDetectionCount": 55,
            },
            "stageMode": "INTERACTIVE",
            "startTime": 1622053587,
            "status": "IN_PROGRESS",
            "storageTime": 1622057607,
          }
        }
        key="1622053587 AL1"
      >
        <TooltipWrapper
          content={
            <TooltipPanel
              activeAnalysts="larry, moe, curly"
              isStale={false}
              lastModified="2021-05-26 19:34:27"
              status="In Progress"
            />
          }
        >
          <Blueprint3.Tooltip
            boundary="window"
            className="core-tooltip"
            content={
              <TooltipPanel
                activeAnalysts="larry, moe, curly"
                isStale={false}
                lastModified="2021-05-26 19:34:27"
                status="In Progress"
              />
            }
            hoverCloseDelay={0}
            hoverOpenDelay={300}
            position="bottom"
            targetClassName="core-tooltip__target"
            targetTagName="div"
            transitionDuration={100}
            wrapperTagName="div"
          >
            <Blueprint3.Popover
              autoFocus={false}
              boundary="window"
              canEscapeKeyClose={false}
              captureDismiss={false}
              className="core-tooltip"
              content={
                <TooltipPanel
                  activeAnalysts="larry, moe, curly"
                  isStale={false}
                  lastModified="2021-05-26 19:34:27"
                  status="In Progress"
                />
              }
              defaultIsOpen={false}
              disabled={false}
              enforceFocus={false}
              fill={false}
              hasBackdrop={false}
              hoverCloseDelay={0}
              hoverOpenDelay={300}
              inheritDarkTheme={true}
              interactionKind="hover-target"
              lazy={true}
              minimal={false}
              modifiers={Object {}}
              openOnTargetFocus={true}
              popoverClassName="bp3-tooltip"
              position="bottom"
              targetClassName="core-tooltip__target"
              targetTagName="div"
              transitionDuration={100}
              usePortal={true}
              wrapperTagName="div"
            >
              <Manager>
                <div
                  className="bp3-popover-wrapper core-tooltip"
                >
                  <Reference
                    innerRef={[Function]}
                  >
                    <InnerReference
                      innerRef={[Function]}
                      setReferenceNode={[Function]}
                    >
                      <Blueprint3.ResizeSensor
                        onResize={[Function]}
                      >
                        <div
                          className="bp3-popover-target core-tooltip__target"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <div
                            className="internal-row"
                            key=".0"
                            tabIndex={0}
                          >
                            <Memo()
                              stageInterval={
                                Object {
                                  "activityIntervals": Array [
                                    Object {
                                      "activeAnalysts": Array [
                                        "larry",
                                        "moe",
                                        "curly",
                                      ],
                                      "comment": "interval example",
                                      "endTime": 1622057187,
                                      "modificationTime": 1622057667,
                                      "name": "Event Review",
                                      "percentAvailable": 100,
                                      "processingEndTime": 1622057547,
                                      "processingStartTime": 1622057487,
                                      "stageName": "AL1",
                                      "startTime": 1622053587,
                                      "status": "IN_PROGRESS",
                                      "storageTime": 1622057607,
                                    },
                                  ],
                                  "comment": "interval example",
                                  "endTime": 1622057187,
                                  "modificationTime": 1622057667,
                                  "name": "AL1",
                                  "percentAvailable": 100,
                                  "processingEndTime": 1622057547,
                                  "processingStartTime": 1622057487,
                                  "stageMetrics": Object {
                                    "associatedSignalDetectionCount": 34,
                                    "eventCount": 21,
                                    "maxMagnitude": 8,
                                    "unassociatedSignalDetectionCount": 55,
                                  },
                                  "stageMode": "INTERACTIVE",
                                  "startTime": 1622053587,
                                  "status": "IN_PROGRESS",
                                  "storageTime": 1622057607,
                                }
                              }
                            >
                              <Memo()
                                key="AL1"
                                stageInterval={
                                  Object {
                                    "activityIntervals": Array [
                                      Object {
                                        "activeAnalysts": Array [
                                          "larry",
                                          "moe",
                                          "curly",
                                        ],
                                        "comment": "interval example",
                                        "endTime": 1622057187,
                                        "modificationTime": 1622057667,
                                        "name": "Event Review",
                                        "percentAvailable": 100,
                                        "processingEndTime": 1622057547,
                                        "processingStartTime": 1622057487,
                                        "stageName": "AL1",
                                        "startTime": 1622053587,
                                        "status": "IN_PROGRESS",
                                        "storageTime": 1622057607,
                                      },
                                    ],
                                    "comment": "interval example",
                                    "endTime": 1622057187,
                                    "modificationTime": 1622057667,
                                    "name": "AL1",
                                    "percentAvailable": 100,
                                    "processingEndTime": 1622057547,
                                    "processingStartTime": 1622057487,
                                    "stageMetrics": Object {
                                      "associatedSignalDetectionCount": 34,
                                      "eventCount": 21,
                                      "maxMagnitude": 8,
                                      "unassociatedSignalDetectionCount": 55,
                                    },
                                    "stageMode": "INTERACTIVE",
                                    "startTime": 1622053587,
                                    "status": "IN_PROGRESS",
                                    "storageTime": 1622057607,
                                  }
                                }
                              >
                                <div
                                  className="interval-cell interval-cell--in-progress"
                                  key="1622053587"
                                  onContextMenu={[Function]}
                                  style={
                                    Object {
                                      "gridColumn": "span 12",
                                    }
                                  }
                                />
                              </Memo()>
                            </Memo()>
                          </div>
                        </div>
                      </Blueprint3.ResizeSensor>
                    </InnerReference>
                  </Reference>
                  <Blueprint3.Overlay
                    autoFocus={false}
                    backdropClassName="bp3-popover-backdrop"
                    backdropProps={Object {}}
                    canEscapeKeyClose={false}
                    canOutsideClickClose={false}
                    enforceFocus={false}
                    hasBackdrop={false}
                    isOpen={false}
                    lazy={true}
                    onClose={[Function]}
                    transitionDuration={100}
                    transitionName="bp3-popover"
                    usePortal={true}
                  />
                </div>
              </Manager>
            </Blueprint3.Popover>
          </Blueprint3.Tooltip>
        </TooltipWrapper>
      </Tooltip>
    </div>
  </WorkflowTableStage>
</Provider>
`;

exports[`Workflow Table Stage mounts with automatic processing interval 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <WorkflowTableStage
    isExpanded={false}
    stage={
      Object {
        "comment": "interval example",
        "endTime": 1622057187,
        "modificationTime": 1622057667,
        "name": "Auto Network",
        "percentAvailable": 100,
        "processingEndTime": 1622057547,
        "processingStartTime": 1622057487,
        "sequenceIntervals": Array [
          Object {
            "comment": "interval example",
            "endTime": 1622057187,
            "lastExecutedStepName": "last-step",
            "modificationTime": 1622057667,
            "name": "Auto Network Seq",
            "percentAvailable": 100,
            "percentComplete": 50,
            "processingEndTime": 1622057547,
            "processingStartTime": 1622057487,
            "stageName": "Auto Network",
            "startTime": 1622053587,
            "status": "IN_PROGRESS",
            "storageTime": 1622057607,
          },
        ],
        "stageMetrics": Object {
          "associatedSignalDetectionCount": 34,
          "eventCount": 21,
          "maxMagnitude": 8,
          "unassociatedSignalDetectionCount": 55,
        },
        "stageMode": "AUTOMATIC",
        "startTime": 1622053587,
        "status": "IN_PROGRESS",
        "storageTime": 1622057607,
      }
    }
    subRowNames={Array []}
  >
    <div
      className="stage-column"
    >
      <Tooltip
        interval={
          Object {
            "comment": "interval example",
            "endTime": 1622057187,
            "modificationTime": 1622057667,
            "name": "Auto Network",
            "percentAvailable": 100,
            "processingEndTime": 1622057547,
            "processingStartTime": 1622057487,
            "sequenceIntervals": Array [
              Object {
                "comment": "interval example",
                "endTime": 1622057187,
                "lastExecutedStepName": "last-step",
                "modificationTime": 1622057667,
                "name": "Auto Network Seq",
                "percentAvailable": 100,
                "percentComplete": 50,
                "processingEndTime": 1622057547,
                "processingStartTime": 1622057487,
                "stageName": "Auto Network",
                "startTime": 1622053587,
                "status": "IN_PROGRESS",
                "storageTime": 1622057607,
              },
            ],
            "stageMetrics": Object {
              "associatedSignalDetectionCount": 34,
              "eventCount": 21,
              "maxMagnitude": 8,
              "unassociatedSignalDetectionCount": 55,
            },
            "stageMode": "AUTOMATIC",
            "startTime": 1622053587,
            "status": "IN_PROGRESS",
            "storageTime": 1622057607,
          }
        }
        key="1622053587 Auto Network"
      >
        <TooltipWrapper
          content={
            <TooltipPanel
              isStale={false}
              lastModified="2021-05-26 19:34:27"
              status="In Progress"
            />
          }
        >
          <Blueprint3.Tooltip
            boundary="window"
            className="core-tooltip"
            content={
              <TooltipPanel
                isStale={false}
                lastModified="2021-05-26 19:34:27"
                status="In Progress"
              />
            }
            hoverCloseDelay={0}
            hoverOpenDelay={300}
            position="bottom"
            targetClassName="core-tooltip__target"
            targetTagName="div"
            transitionDuration={100}
            wrapperTagName="div"
          >
            <Blueprint3.Popover
              autoFocus={false}
              boundary="window"
              canEscapeKeyClose={false}
              captureDismiss={false}
              className="core-tooltip"
              content={
                <TooltipPanel
                  isStale={false}
                  lastModified="2021-05-26 19:34:27"
                  status="In Progress"
                />
              }
              defaultIsOpen={false}
              disabled={false}
              enforceFocus={false}
              fill={false}
              hasBackdrop={false}
              hoverCloseDelay={0}
              hoverOpenDelay={300}
              inheritDarkTheme={true}
              interactionKind="hover-target"
              lazy={true}
              minimal={false}
              modifiers={Object {}}
              openOnTargetFocus={true}
              popoverClassName="bp3-tooltip"
              position="bottom"
              targetClassName="core-tooltip__target"
              targetTagName="div"
              transitionDuration={100}
              usePortal={true}
              wrapperTagName="div"
            >
              <Manager>
                <div
                  className="bp3-popover-wrapper core-tooltip"
                >
                  <Reference
                    innerRef={[Function]}
                  >
                    <InnerReference
                      innerRef={[Function]}
                      setReferenceNode={[Function]}
                    >
                      <Blueprint3.ResizeSensor
                        onResize={[Function]}
                      >
                        <div
                          className="bp3-popover-target core-tooltip__target"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <div
                            className="internal-row"
                            key=".0"
                            tabIndex={0}
                          >
                            <Memo()
                              stageInterval={
                                Object {
                                  "comment": "interval example",
                                  "endTime": 1622057187,
                                  "modificationTime": 1622057667,
                                  "name": "Auto Network",
                                  "percentAvailable": 100,
                                  "processingEndTime": 1622057547,
                                  "processingStartTime": 1622057487,
                                  "sequenceIntervals": Array [
                                    Object {
                                      "comment": "interval example",
                                      "endTime": 1622057187,
                                      "lastExecutedStepName": "last-step",
                                      "modificationTime": 1622057667,
                                      "name": "Auto Network Seq",
                                      "percentAvailable": 100,
                                      "percentComplete": 50,
                                      "processingEndTime": 1622057547,
                                      "processingStartTime": 1622057487,
                                      "stageName": "Auto Network",
                                      "startTime": 1622053587,
                                      "status": "IN_PROGRESS",
                                      "storageTime": 1622057607,
                                    },
                                  ],
                                  "stageMetrics": Object {
                                    "associatedSignalDetectionCount": 34,
                                    "eventCount": 21,
                                    "maxMagnitude": 8,
                                    "unassociatedSignalDetectionCount": 55,
                                  },
                                  "stageMode": "AUTOMATIC",
                                  "startTime": 1622053587,
                                  "status": "IN_PROGRESS",
                                  "storageTime": 1622057607,
                                }
                              }
                            >
                              <Memo()
                                key="Auto Network"
                                stageInterval={
                                  Object {
                                    "comment": "interval example",
                                    "endTime": 1622057187,
                                    "modificationTime": 1622057667,
                                    "name": "Auto Network",
                                    "percentAvailable": 100,
                                    "processingEndTime": 1622057547,
                                    "processingStartTime": 1622057487,
                                    "sequenceIntervals": Array [
                                      Object {
                                        "comment": "interval example",
                                        "endTime": 1622057187,
                                        "lastExecutedStepName": "last-step",
                                        "modificationTime": 1622057667,
                                        "name": "Auto Network Seq",
                                        "percentAvailable": 100,
                                        "percentComplete": 50,
                                        "processingEndTime": 1622057547,
                                        "processingStartTime": 1622057487,
                                        "stageName": "Auto Network",
                                        "startTime": 1622053587,
                                        "status": "IN_PROGRESS",
                                        "storageTime": 1622057607,
                                      },
                                    ],
                                    "stageMetrics": Object {
                                      "associatedSignalDetectionCount": 34,
                                      "eventCount": 21,
                                      "maxMagnitude": 8,
                                      "unassociatedSignalDetectionCount": 55,
                                    },
                                    "stageMode": "AUTOMATIC",
                                    "startTime": 1622053587,
                                    "status": "IN_PROGRESS",
                                    "storageTime": 1622057607,
                                  }
                                }
                              >
                                <div
                                  className="interval-cell interval-cell--in-progress"
                                  onContextMenu={[Function]}
                                  style={
                                    Object {
                                      "gridColumn": "span 12",
                                    }
                                  }
                                >
                                  <PercentBar
                                    percentage={100}
                                  >
                                    <div
                                      className="percent-bar"
                                      key="1"
                                      style={
                                        Object {
                                          "transform": "translateX(-0%)",
                                          "width": "100%",
                                        }
                                      }
                                    />
                                  </PercentBar>
                                </div>
                              </Memo()>
                            </Memo()>
                          </div>
                        </div>
                      </Blueprint3.ResizeSensor>
                    </InnerReference>
                  </Reference>
                  <Blueprint3.Overlay
                    autoFocus={false}
                    backdropClassName="bp3-popover-backdrop"
                    backdropProps={Object {}}
                    canEscapeKeyClose={false}
                    canOutsideClickClose={false}
                    enforceFocus={false}
                    hasBackdrop={false}
                    isOpen={false}
                    lazy={true}
                    onClose={[Function]}
                    transitionDuration={100}
                    transitionName="bp3-popover"
                    usePortal={true}
                  />
                </div>
              </Manager>
            </Blueprint3.Popover>
          </Blueprint3.Tooltip>
        </TooltipWrapper>
      </Tooltip>
    </div>
  </WorkflowTableStage>
   
</Provider>
`;

exports[`Workflow Table Stage shallow mounts 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ContextProvider
    value={
      Object {
        "allActivitiesOpenForSelectedInterval": false,
        "closeConfirmationPrompt": [MockFunction],
        "openAnythingConfirmationPrompt": [MockFunction],
        "openConfirmationPrompt": [MockFunction],
        "staleStartTime": 1,
      }
    }
  >
    <WorkflowTableStage
      isExpanded={false}
      stage={
        Object {
          "activityIntervals": Array [
            Object {
              "activeAnalysts": Array [
                "larry",
                "moe",
                "curly",
              ],
              "comment": "interval example",
              "endTime": 1622057187,
              "modificationTime": 1622057667,
              "name": "Event Review",
              "percentAvailable": 100,
              "processingEndTime": 1622057547,
              "processingStartTime": 1622057487,
              "stageName": "AL1",
              "startTime": 1622053587,
              "status": "IN_PROGRESS",
              "storageTime": 1622057607,
            },
          ],
          "comment": "interval example",
          "endTime": 1622057187,
          "modificationTime": 1622057667,
          "name": "AL1",
          "percentAvailable": 100,
          "processingEndTime": 1622057547,
          "processingStartTime": 1622057487,
          "stageMetrics": Object {
            "associatedSignalDetectionCount": 34,
            "eventCount": 21,
            "maxMagnitude": 8,
            "unassociatedSignalDetectionCount": 55,
          },
          "stageMode": "INTERACTIVE",
          "startTime": 1622053587,
          "status": "IN_PROGRESS",
          "storageTime": 1622057607,
        }
      }
      subRowNames={Array []}
    />
     
  </ContextProvider>
</ContextProvider>
`;
